src/main/g8/test/forms/behaviours/FormBehaviours.scala

```diff
@@ -55,6 +55,34 @@ trait FormBehaviours extends FormSpec {
       checkForError(form, data, expectedError)
     }
   }
+  
+  def formWithMaxLengthTextFields(fields: MaxLengthField*) = {
+    for (field <- fields) {
+      s"fail to bind when ${field.fieldName} has more characters than ${field.maxLength}" in {
+        val invalid = "A" * (field.maxLength + 1)
+        val validFields = validData - field.fieldName
+        val data = validFields ++ Map(field.fieldName -> invalid)
+        val expectedError = error(field.fieldName, field.errorMessageKey, field.maxLength)
+        checkForError(form, data, expectedError)
+      }
+    }
+  }
+
+  def formWithMandatoryTextFieldsAndCustomKey(fields: (String, String)*) = {
+    for ((key, errorMessage) <- fields) {
+      s"fail to bind when $key is omitted" in {
+        val data = validData - key + (key -> "")
```

Comment:
No need for the `- key` in this line, since adding an existing key to a Map will overwrite it.